waterbox/bsnescore/bsnes/sfc/ppu-fast/ppu.cpp

```diff
@@ -52,13 +54,21 @@ PPU::PPU() {
   }
 
   for(uint y : range(240)) {
+    lines[y].cgram = (uint16_t*)alloc_invisible(256 * sizeof (uint16_t));
+    lines[y].items = (ObjectItem*)alloc_invisible(128 * sizeof (ObjectItem));
+    lines[y].tiles = (ObjectTile*)alloc_invisible(128 * sizeof (ObjectTile));
+    lines[y].above = (Pixel*)alloc_invisible(256 * 9 * 9 * sizeof (Pixel));
+    lines[y].below = (Pixel*)alloc_invisible(256 * 9 * 9 * sizeof (Pixel));
+    lines[y].windowAbove = (bool*)alloc_invisible(256 * sizeof (bool));
+    lines[y].windowBelow = (bool*)alloc_invisible(256 * sizeof (bool));
```

Comment:
Consider using the C++ version `alloc_invisible<T>(size)` instead of `(T*) alloc_invisible(size * sizeof(T))` (here and in all places where it makes sense)

Reply:
I think the bsnes 87 core predates the C++ typed overloads on the alloc functions, which is why you wouldn't have seen them there.

Reply:
Ah, I was largely looking at old bsnes core for clues so didn't know c++ got some nice options for this.